### PR TITLE
Allow 'match' operator

### LINKED ISSUE #3359: The operator "MATCH" is not permitted
# Environment

Knex version: 0.15.2
Database + version: SQLite 3.28
OS: Arch Linux

Select applicable template from below.
If issue is about oracledb support, tag @ atiertant. For MSSql tag @ smorey2.
Rest of dialects don't need tags.
If issue is about TypeScript definitions, tag @ lorefnon.

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

I'm trying to run a query on an FTS table. It's something like this:

``` javascript
const findProducts = (queryString, limit) => {
  return knex
      .select('*)
      .from('productos')
      .join('productosFts', { 'productos.ftsid': 'productosFts.rowid' })
      .where('productosFts.nombre', 'MATCH', queryString + '*')
      .limit(limit);
}
```

2. Error message

This query throws the following error:

``` javascript
TypeError: The operator "MATCH" is not permitted
        at Formatter.operator (/home/gabriel/projects/Facturacion/backend/node_modules/knex/lib/formatter.js:187:13)
        at QueryCompiler_SQLite3.whereBasic (/home/gabriel/projects/Facturacion/backend/node_modules/knex/lib/query/compiler.js:554:100)
        at QueryCompiler_SQLite3.where (/home/gabriel/projects/Facturacion/backend/node_modules/knex/lib/query/compiler.js:350:32)
        at /home/gabriel/projects/Facturacion/backend/node_modules/knex/lib/query/compiler.js:127:31
        at Array.map (<anonymous>)
        at QueryCompiler_SQLite3.select (/home/gabriel/projects/Facturacion/backend/node_modules/knex/lib/query/compiler.js:126:33)
        at QueryCompiler_SQLite3.toSQL (/home/gabriel/projects/Facturacion/backend/node_modules/knex/lib/query/compiler.js:74:27)
        at Builder.toSQL (/home/gabriel/projects/Facturacion/backend/node_modules/knex/lib/query/builder.js:70:44)
        at /home/gabriel/projects/Facturacion/backend/node_modules/knex/lib/runner.js:39:32
        at tryCatcher (/home/gabriel/projects/Facturacion/backend/node_modules/bluebird/js/release/util.js:16:23)
        at /home/gabriel/projects/Facturacion/backend/node_modules/bluebird/js/release/using.js:185:26
        at tryCatcher (/home/gabriel/projects/Facturacion/backend/node_modules/bluebird/js/release/util.js:16:23)
        at Promise._settlePromiseFromHandler (/home/gabriel/projects/Facturacion/backend/node_modules/bluebird/js/release/promise.js:512:31)
        at Promise._settlePromise (/home/gabriel/projects/Facturacion/backend/node_modules/bluebird/js/release/promise.js:569:18)
        at Promise._settlePromise0 (/home/gabriel/projects/Facturacion/backend/node_modules/bluebird/js/release/promise.js:614:10)
        at Promise._settlePromises (/home/gabriel/projects/Facturacion/backend/node_modules/bluebird/js/release/promise.js:694:18)
```

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

Any query using `MATCH` should throw the same error.

---

Adding `match` to the operators array in `lib/formatter.js` is a quick workaround. I already tested it on my codebase by editing the code in `node_modules`. Works great for my use case but I'm not sure if this is the best solution though.

### PR DESCRIPTION
**Pull request by GAumala**
Closes #3359

### CODE REVIEW

**Comment by kibertoad** [comment]
@GAumala Do you need any help with this?

**Comment by GAumala** [comment]
Yes, I'm not sure how should I write a test for this. It doesn't seem very simple because:

-  It would have to be SQLite only, I don't think other databases support `MATCH`. 
- It would require a new table created with fts.

**Comment by kibertoad** [comment]
@GAumala Check test/integration2/schema/foreign-keys.spec.js for an example of how to do DB-specific tests. You can skip everything that `!isSQLite(knex))`

**Comment by GAumala** [comment]
Thanks for the help! I added a test. Let me know if further changes are needed.

File: test/integration2/query/select/fts.spec.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        'CREATE VIRTUAL TABLE fts_products USING fts5(name);'
+      );
+    });
+
+    afterEach(async () => {
+      await knex.schema.dropTable('fts_products');
+    });
+
+    it('selects from fts tables using "match" operator', async function () {
+      if (!isSQLite(knex)) {
```
Comment (kibertoad) [inline test/integration2/query/select/fts.spec.js:28]: considering that you only retrieve sqlite config, this seems redundant.

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Thank you!

**Pull request merged**